tests/test_metrics.py

```diff
@@ -385,5 +387,30 @@ def test_classifier_metrics(self):
         assert agg['weighted_f1'] == (0.5 + (2 / 3) * 0.5 + (4 / 7) * 0.5) / 2
 
 
+class TestDistinct(unittest.TestCase):
+    def test_inter_distinct(self):
+        # 3 n-grams, all appearing once
+        m = InterDistinctMetric.compute("this is some test", 2)
+        self.assertAlmostEqual(m, 1.0)
+        # 3-grams, each appearing twice
+        self.assertAlmostEqual(m + m, 0.5)
+
+    def test_inter_distinct_unigram(self):
+        m1 = InterDistinctMetric.compute("this test", 1)
+        self.assertAlmostEqual(m1, 1.0, delta=0.001)
+        m2 = InterDistinctMetric.compute("another test", 1)
+        self.assertAlmostEqual(m2, 1.0, delta=0.001)
+        # we now have 4 tokens, 3 words
+        self.assertAlmostEqual(m1 + m2, 3 / 4)
+
+    def test_intra_distinct(self):
+        # 4/5 are unique
+        m1 = IntraDistinctMetric.compute("this is some test test", 1)
```

Comment:
Do you think it makes sense to iteratively compute this metrics on multiple examples (averaged over multiple ones)

Reply:
I mean I do... I have my base case and my inductive case

Reply:
oh I see you have such tests for L404 and 412. No more questions from me.